pkg/credentialprovider/plugin/config.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"k8s.io/kubernetes/pkg/credentialprovider/apis/criauth"
+)
+
+// readCRIAuthPluginConfigFile receives a path to a config file and decodes it
+// into the internal CRIAuthPluginConfig type.
+func readCRIAuthPluginConfigFile(configPath string) (*criauth.CRIAuthPluginConfig, error) {
+	if configPath == "" {
+		return nil, fmt.Errorf("CRI auth plugin config path is empty")
+	}
+
+	data, err := ioutil.ReadFile(configPath)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read external registry credential provider configuration from %q [%v]", configPath, err)
+	}
+
+	return decode(data)
+}
+
+// decode decodes data into the internal CRIAuthPluginConfig type.
+func decode(data []byte) (*criauth.CRIAuthPluginConfig, error) {
+	obj, gvk, err := codecs.UniversalDecoder().Decode(data, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if gvk.Kind != "CRIAuthPluginConfig" {
+		return nil, fmt.Errorf("failed to decode %q (missing Kind)", gvk.Kind)
+	}
+
+	if internalConfig, ok := obj.(*criauth.CRIAuthPluginConfig); ok {
+		return internalConfig, nil
+	}
+
+	return nil, fmt.Errorf("unable to convert %T to *CRIAuthPluginConfig", obj)
```

Comment:
nil: should the actual decoding error be surfaced here? May be informative

Reply:
I'm assuming not based on your other comment here https://github.com/kubernetes/kubernetes/pull/94196#discussion_r493094993

Reply:
there are two places parsing happens - config and response. I agree about the response. But config does not contain credentials and cluster admin would benefit from more informative error messages from the configuration parsing